projects/cashmere/src/lib/stepper/stepper.component.ts

```diff
@@ -0,0 +1,135 @@
+import {Router, NavigationEnd} from '@angular/router';
+import {parseBooleanAttribute} from '../util';
+
+export function throwErrorForMissingRouterLink(stepsWithoutRouterLink: StepInterface[]) {
+    const stepLabels = stepsWithoutRouterLink.map(step => step.label);
+    throw Error(`Routerlink missing on ${stepLabels.join(',')}`);
+}
+
+export interface StepInterface {
+    /** The text to be displayed under the step indicator */
+    label?: string;
+    /** If using a router, the route this step should navigate to */
+    routerLink?: string;
+    /** The icon set the step's hc-icon should use. Must be set if using the icon parameter. */
+    iconSet?: string;
+    /** An glyph that should be displayed inside the step's circle.
+     * If not set, the step number will display in the circle instead. */
+    icon?: string;
+    /** If true, the step will not be clickable */
+    disabled?: boolean;
+}
+
+/** Convey progress through numbered steps, providing a wizard-like workflow.  */
+@Component({
+    selector: 'hc-stepper',
+    templateUrl: './stepper.component.html',
+    styleUrls: ['./stepper.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class StepperComponent implements AfterContentInit {
+    _routerEnabled = false;
+
+    /** An array defining the steps in the stepper */
+    @Input()
+    public steps: StepInterface[];
+
+    /** Sets the layout of the progress stepper. *Defaults to `arrow`.*  */
+    @Input()
+    public type: 'arrow' | 'isolated' = 'arrow';
+
+    /** Sets the highlight color of the progress stepper. *Defaults to `green`.*  */
+    @Input()
+    get color(): 'green' | 'blue' | 'purple' | 'orange' | 'red' | 'none' {
+        return this._color;
+    }
+    set color(value: 'green' | 'blue' | 'purple' | 'orange' | 'red' | 'none') {
+        this._color = value;
+        this._hostClass = 'hc-stepper-' + this._color;
+    }
+    private _color: 'green' | 'blue' | 'purple' | 'orange' | 'red' | 'none' = 'green';
+
+    /** Determines whether numerals should be displayed on each step indicator for the isolated type.
+     * *Defaults to `false`.*  */
+    @Input()
+    get showStepCount(): boolean {
+        return this._showStepCount;
+    }
+    set showStepCount(value) {
+        this._showStepCount = parseBooleanAttribute(value);
+    }
+    private _showStepCount: boolean = false;
+
+    /** If true, include a router outlet with the component. *Defaults to `true`.*  */
+    @Input()
+    get useRouterOutlet(): boolean {
+        return this._useRouterOutlet;
+    }
+    set useRouterOutlet(value) {
+        this._useRouterOutlet = parseBooleanAttribute(value);
+    }
+    private _useRouterOutlet: boolean = true;
+
+    /** Get or set the currently selected zero-based index of the stepper */
+    @Input()
+    get activeIndex(): number {
+        return this._activeIndex;
+    }
+    set activeIndex(value: number) {
+        if (this.steps) {
```

Comment:
A nitpick, but let's invert this `if` statement with an early return.

Reply:
Got it - so like this?
```
set activeIndex(value: number) {
        if (!this.steps) {
            return;
        } else if (value < 0 || value >= this.steps.length) {
            throw Error('The hc-stepper activeIndex value of ' + value + ' is out of bounds');
        } else {
            if (this._routerEnabled) {
                this.router.navigate([this.steps[value].routerLink]);
            } else {
                this._activeIndex = value;
                this.activeIndexChange.emit(this._activeIndex);
            }
        }
    }
```

Reply:
Almost. More like this:

```javascript
if (!this.steps) {
  return;
}

if (value < 0 || value >= this.steps.length) {
  throw Error(`...`);
}

if (this._routerEnabled) {
  this.router.navigate([this.steps[value].routerLink]);
} else {
  this._activeIndex = value;
  this.activeIndexChange.emit(this._activeIndex);
}
```

The goal is to reduce logical nesting and therefore cyclomatic complexity.

Reply:
Fascinating.  I confess that's the first time I've heard the term "cyclomatic complexity".  But I learned something new today :-) thanks Issac

Reply:
ha, no problem. It's probably the most pretentious term I know

Reply:
Haha - it's good one! Fixed in the most recent commit.